api/api.go

```diff
@@ -193,75 +184,71 @@ func (b *BlockChainAPI) GetTransactionByBlockHashAndIndex(
 	ctx context.Context,
 	blockHash common.Hash,
 	index hexutil.Uint,
-) *RPCTransaction { // todo add error to the return type
+) (*RPCTransaction, error) { // todo add error to the return type
 	block, err := b.blocks.GetByID(blockHash)
 	if err != nil {
-		b.logger.Error().Err(err).Msg("failed to get block by id")
-		return nil
+		return handleError[*RPCTransaction](b.logger, err)
 	}
 
 	highestIndex := len(block.TransactionHashes) - 1
 	if index > hexutil.Uint(highestIndex) {
-		return nil
+		return nil, nil
 	}
 
 	txHash := block.TransactionHashes[index]
 	tx, err := b.GetTransactionByHash(ctx, txHash)
 	if err != nil {
-		b.logger.Error().Err(err).Msg("failed to get transaction by id")
-		return nil
+		return handleError[*RPCTransaction](b.logger, err)
 	}
 
-	return tx
+	return tx, nil
 }
 
 // GetTransactionByBlockNumberAndIndex returns the transaction for the given block number and index.
 func (b *BlockChainAPI) GetTransactionByBlockNumberAndIndex(
 	ctx context.Context,
 	blockNumber rpc.BlockNumber,
 	index hexutil.Uint,
-) *RPCTransaction { // todo add error to the return type
+) (*RPCTransaction, error) { // todo add error to the return type
 	block, err := b.blocks.GetByHeight(uint64(blockNumber))
 	if err != nil {
-		if !errors.Is(err, storageErrs.NotFound) {
-			b.logger.Error().Err(err).Msg("failed to get block by height")
-		}
-		return nil
+		return handleError[*RPCTransaction](b.logger, err)
 	}
 
 	highestIndex := len(block.TransactionHashes) - 1
 	if index > hexutil.Uint(highestIndex) {
-		return nil
+		return nil, nil
```

Comment:
I guess we should return some kind of error here, instead of both values being nil, nil.

Reply:
This is indicating a not found response. It took me some time to figure out but JSON RPC clients expect nil nil in case the resource is not found.